styx-common/src/main/java/com/spotify/styx/publisher/EventInterceptor.java

```diff
@@ -0,0 +1,40 @@
+/*-
+ * -\-\-
+ * Spotify Styx Common
+ * --
+ * Copyright (C) 2016 - 2017 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.styx.publisher;
+
+import com.spotify.styx.model.SequenceEvent;
+
+/**
+ * Interface for acting on internal Styx events. For example, this can be used to publish internal
+ * Styx events to third-party databases.
+ */
+public interface EventInterceptor {
+
+  /**
+   * Called when a state machine transition happens due to a new internal Styx event.
```

Comment:
"Called _when_ transition happens" is uninformative. It's actually called _after_ I guess? not before? and not concurrently? Also `interceptedEvent` is uninformative in the same manner. We could reflect more exact semantics in the name easily.